src/quat.rs

```diff
@@ -128,6 +128,45 @@ macro_rules! impl_quat_methods {
             ))
         }
 
+        /// Gets the minimal rotation for transforming `from` to `to`.
+        /// The rotation is in the plane spanned by the two vectors.
+        /// Will rotate at most 180 degrees.
+        ///
+        /// The input vectors must be finite and non-zero.
+        ///
+        /// `from_rotation_arc(from, to) * from ≈ to`.
+        pub fn from_rotation_arc(from: $vec3, to: $vec3) -> Self {
+            let from = from.normalize();
+            let to = to.normalize();
```

Comment:
Typically in `glam` I have avoided normalizing input vectors and instead expect the input to be pre-normalized. I have `glam_assert` which can be enabled to check that parameters are valid if necessary. I try and comment where inputs need to meet some requirement like this although sometimes I've forgetten to add a comment (e.g. Quat::from_axis_angle`. I work on the premise that people shouldn't pay the cost if their data is already normalized. Often calling code may also need to normalize their vectors for whatever else they are doing, so I figure if I require parameters to be normalized then it won't happen twice.

I suppose in this case it would be reasonable for vectors to not necessarily be unit length...

Alternatively, this could optionally normalize if the input is not already normalized, which at least would avoid normalizing already normalized data.

Reply:
I think your design choice "pay for what you use" makes sense. In this case I think it is reasonable to expect users to normalize themselves. I'll replace the normalization with `glam_assert` and a docstring!